python/tests/test_model_load.py

```diff
@@ -0,0 +1,52 @@
+import json
+import re
+from pathlib import Path
+
+from git import Repo
+from packaging import version
+from pytest import fixture
+
+from _messages import Envelope
+
+
+@fixture
+def compatibility_kit_repo(tmpdir):
+    repo_path = Path(tmpdir) / "compatibility-kit"
+    repo = Repo.clone_from(
+        "https://github.com/cucumber/compatibility-kit.git",
```

Comment:
Messages should not use the compatibility kit as this creates a circular dependency. Rather you'll want to write some targeted tests for serialization and deserialisation.

Reply:
@luke-hill the above comment also applies to you.

Reply:
Thanks! The small suite of tests will be copied here.
Java tests seem must not be directly ported because the model is generated from schemas directly. So many tests will just test the generator itself(it has a much wider suite of tests)

Reply:
Indeed. Most of the tests in Java are for serialization rather than the shape of the messages.

For example enums must be serialized by name,  null fields must be omitted, ect. This will depend a bit on what Python offers out of the box.

Reply:
It's worth pointing out here @elchupanebrej the way in which Rien is describing things is that we should / can use the CCK to test the generation. But we shouldn't have the generation "requiring on" the CCK. Hope that makes sense / apologies if I'm repeating something already understood.

i.e. for ruby here - https://github.com/cucumber/messages/blob/main/ruby/cucumber-messages.gemspec we have no direct dependencies, but we use the CCK as a development dependency (I.e. to test the generation has worked).

Apologies if this doesn't make sense.

Reply:
PyPi releases don't allow dependencies on github repositories, so I can't  add resources directly. If you go through commits you will see an example of tests with direct downloading CCK data. If you have better ideas how to integrate - please share your thoughts

Reply:
Surely this isn't a problem if it's listed as a dev dependency as it won't be in the pypi package?

Reply:
@luke-hill, this implementation isn't dependent on CCK for generation, it was dependent on CCK for test purposes only